vault/core.go

```diff
@@ -2141,6 +2163,16 @@ func (s standardUnsealStrategy) unseal(ctx context.Context, logger log.Logger, c
 	// for the active startup.
 	c.activeTime = time.Now().UTC()
 
+	if err := c.loadVersionHistory(ctx); err != nil {
+		return err
+	}
+
+	var err error
+	c.majorVersionFirstMount, err = c.isMajorVersionFirstMount(ctx)
```

Comment:
In principle this is fine, since we never call setupCredentials/setupMounts unless we get past this line.  But I don't really like it when a func returns (something,error) and the something is assigned to a non-locally-scoped value before we check to see if the error is nil.  It's the kind of thing that ideally I'd like to scan for as a potential bug pattern, e.g. with semgrep.

Reply:
No problem, I've changed this locally.